test/runnable/testinvariant.d

```diff
@@ -25,6 +25,76 @@ int testinvariant()
     return 0;
 }
 
+/***************************************************/
+// 519
+
+class C519
+{
+    invariant
+    {
+        ++x;
+    }
+    __gshared int x;
+}
+
+struct S519
+{
+    invariant()
+    {
+        printf("S519.invariant\n");
+        ++x;
+    }
+    __gshared int x;
+}
+
+struct S519c
+{
+    invariant()
+    {
+        printf("S519c.invariant\n");
+        ++x;
+    }
+    __gshared int x;
+    int y;
+}
+
+void test519()
+{
+    //printf("test1\n");
+    {
+        auto s = new C519();
+        assert(C519.x == 0);
+        delete s;
+        assert(C519.x == 1);
+    }
+
+    //printf("test2\n");
+    {
+        auto foo = new S519();
+        printf("lifetime of foo\n");
+        delete foo;
```

Comment:
`destroy(*foo)` should work instead of `delete foo`

Reply:
Ah, I had forgotten that there's no pointer-to-struct overload for destroy.